clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithCommitFailedExceptionForTimeout() {
```

Comment:
I'm not quite sure how this is supposed to work. The case we're trying to hit is when `client.poll(future, timeout)` returns without the future being completed. A simple way to check this path is something like this:
```java
       client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
       coordinator.ensureCoordinatorReady();
       assertFalse(coordinator.commitOffsetsSync(Collections.singletonMap(t1p, new OffsetAndMetadata(100L)), 0));
 ```
You can confirm that this test fails without the fix above.